PurchasesTests/Caching/DeviceCacheTests.swift

```diff
@@ -194,4 +209,117 @@ class DeviceCacheTests: XCTestCase {
 
         expect { mockNotificationCenter.fireNotifications() }.to(raiseException())
     }
+
+    func testNewDeviceCacheInstanceWithExistingValidPurchaserInfoCacheIsntStale() {
+        let mockNotificationCenter = MockNotificationCenter()
+        let appUserID = "myUser"
+        let fourMinutesAgo = Calendar.current.date(byAdding: .minute, value: -4, to: Date())
+        mockUserDefaults.mockValues["com.revenuecat.userdefaults.purchaserInfoLastUpdated.\(appUserID)"] = fourMinutesAgo
+        self.deviceCache = RCDeviceCache(mockUserDefaults,
+                                         offeringsCachedObject: nil,
+                                         notificationCenter: mockNotificationCenter)
+
+        expect(self.deviceCache.isPurchaserInfoCacheStale(forAppUserID: appUserID,
+                                                          isAppBackgrounded: false)) == false
+    }
```

Comment:
What's the difference between this test and `testIsPurchaserInfoCacheStaleForForeground`. Also, in this test you're calling `mockUserDefaults.mockValues` but in the other tests you are setting it through `setPurchaserInfoCacheTimestamp`

Reply:
I'll take any naming suggestions I can get, but here's what they're supposed to check: 

`testIsPurchaserInfoCacheStaleForForeground`: 
test that the value of isPurchaserInfoCacheStale works correctly when the app is in the foreground. 
- create a cached purchaser info, but stale date. 
- value should be true
- update the purchaserInfo cache date, with a recent value
- value should be false. 


`testNewDeviceCacheInstanceWithExistingValidPurchaserInfoCacheIsntStale` is supposed to ensure that new instances won't return that the value is stale unless it actually is. This is because we previously were storing the lastUpdated value in memory, so this would have had a different result before this PR. 
- store a value for lastUpdated, that's recent
- check that it isn't considered stale

As for using the two mechanisms to mock, they can be unified. I had just forgotten that you could set it directly through the protected method. 
Using both mechanisms might be better coverage? but less consistent. I'm happy to unify.

Reply:
it's ok, both mechanisms work. I wasn't sure I was missing somehting. Thanks for the explanation